target/linux/ath79/dts/qca9557_kt_dw02-412h-64m.dts

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca9557_kt_dw02-412h.dtsi"
+
+/ {
+	model = "KT DW02-412H (64M)";
+	compatible = "kt,dw02-412h-64m", "qca,qca9557";
+};
+
+&nand {
+	status = "okay";
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partition@0 {
+			label = "current";
+			reg = <0x0 0x1000000>;
+		};
+
+		firmware@1000000 {
+			label = "firmware";
+			reg = <0x1000000 0x7000000>;
```

Comment:
firmware _and_ kernel _and_ ubi? You should decide for one approach.

Apart from that, size of the firmware partition is wrong here.

Reply:
Please try multi-level layout like this:
https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/dts/mt7621_netis_wf2881.dts#L71